tests/5.1/order/test_parallel_for_order_reproducible.c

```diff
@@ -0,0 +1,51 @@
+//===------------ test_target_parallel_for_order_reproducible.c -------------===//
+//
+// OpenMP API Version 5.1 Nov 2020
+//
+// This test checks that the order(reproducible) clause is properly handled.
+// Leverages the thread id to determine if the same threads executed in the
+// same order over the array. If both arrays match the test passes, if not
+// it fails as the reproducible clause failed.
+//
+////===----------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+
+int main() {
+	int errors = 0;
+	int x[N];
+	int y[N];
+
+	for (int i = 0; i < N; i++) {
+		x[i] = 0;
+		y[i] = 0;
+	}
+
+	OMPVV_TEST_OFFLOADING;
+
+	#pragma omp target map(tofrom: x,y)
+	{
+
+		#pragma omp parallel for order(concurrent)
+		for (int i = 0; i < N; i++) {
+			x[i] = omp_get_thread_num();	
```

Comment:
This fails with GCC 12 here and below with:
```
test_parallel_for_order_reproducible.c:36:32: error: OpenMP runtime API call ‘omp_get_thread_num’ in a region with ‘order(concurrent)’ clause
   36 |                         x[i] = omp_get_thread_num();
      |                                ^~~~~~~~~~~~~~~~~~~~
```

Reply:
Namely:
>  Restrictions Restrictions to the **order** clause are as follows:
> ...
> *  A region that corresponds to a construct with an **order** clause that specifies **concurrent** may not contain calls to procedures that contain OpenMP directives.

→ https://www.openmp.org/spec-html/5.1/openmpsu47.html#x72-720002.11.3